test/server.js

```diff
@@ -899,6 +916,173 @@ describe("Server", function () {
+      const deployValue = 12345;
+
+      const artifact = require("./testcontracts/FactoryImmutable/Factory.json");
+      const factoryAddress = await deployFromAbiAndBytecode(
+        localWeb3Provider,
+        artifact.abi,
+        artifact.bytecode,
+        accounts[0]
+      );
+
+      // Deploy child by calling deploy(uint)
+      const childMetadata = require("./testcontracts/FactoryImmutable/Child_metadata.json");
+      const childMetadataBuffer = Buffer.from(JSON.stringify(childMetadata));
+      const txReceipt = await callContractMethodWithTx(
+        localWeb3Provider,
+        artifact.abi,
+        factoryAddress,
+        "deploy",
+        accounts[0],
+        [deployValue]
+      );
+
+      const childAddress = txReceipt.events.Deployment.returnValues[0];
+      const sourcePath = path.join(
+        "test",
+        "testcontracts",
+        "FactoryImmutable",
+        "FactoryTest.sol"
+      );
+      const sourceBuffer = fs.readFileSync(sourcePath);
+
+      const abiEncoded = localWeb3Provider.eth.abi.encodeParameter(
+        "uint",
+        deployValue
+      );
+      const res = await chai
+        .request(server.app)
+        .post("/")
+        .field("address", childAddress)
+        .field("chain", defaultContractChain)
+        .field("abiEncodedConstructorArguments", abiEncoded)
+        .attach("files", childMetadataBuffer, "metadata.json")
+        .attach("files", sourceBuffer, "FactoryTest.sol")
+        .send();
+      assertions(null, res, null, childAddress);
+    });
+
+    it.only("should verify a contract created by a factory contract and has immutables without constructor arguments but with msg.sender assigned immutable", async () => {
+      const artifact = require("./testcontracts/FactoryImmutableWithoutConstrArg/Factory3.json");
+      const factoryAddress = await deployFromAbiAndBytecode(
+        localWeb3Provider,
+        artifact.abi,
+        artifact.bytecode,
+        accounts[0]
+      );
+
+      // Deploy child by calling createChild()
+      const childMetadata = require("./testcontracts/FactoryImmutableWithoutConstrArg/Child3_metadata.json");
+      const childMetadataBuffer = Buffer.from(JSON.stringify(childMetadata));
+      const txReceipt = await callContractMethodWithTx(
+        localWeb3Provider,
+        artifact.abi,
+        factoryAddress,
+        "createChild",
+        accounts[0],
+        []
+      );
+
+      const childAddress = txReceipt.events.ChildCreated.returnValues[0];
+      const sourcePath = path.join(
+        "test",
+        "testcontracts",
+        "FactoryImmutableWithoutConstrArg",
+        "FactoryTest3.sol"
+      );
+      const sourceBuffer = fs.readFileSync(sourcePath);
+
+      const res = await chai
+        .request(server.app)
+        .post("/")
+        .field("address", childAddress)
+        .field("chain", defaultContractChain)
+        .field("msgSender", factoryAddress)
```

Comment:
isn't it better to handle the the global variables as an object in which msgSender is one of the properties?

Reply:
(like I suggested you with the stored file on the repo)

Reply:
I agree, also better to keep it consistent with the internal data structure.